vehicle/mkvi/software/bms/bms.c

```diff
@@ -37,10 +37,45 @@ void timer0_isr() {
     run_10ms = true;
 }
 
+void timer1_isr() {
+    timer1_cfg.channel_a.interrupt_enable = false;
+    timer1_cfg.channel_b.output_compare_match = 0;
+    timer_init(&timer1_cfg);
+}
+
 void pcint0_callback() {
     bms_core.bspd_current_sense = !!gpio_get_pin(BSPD_CURRENT_THRESH);
 }
 
+static void control_cooling_loop(void) {
+    static int8_t tolerance = 5; // to be defined
+    static uint16_t current_temp = 0;
+    static uint16_t previous_temp = 0;
+    static uint8_t duty_cycle = 0;
+    static int8_t min_duty = 13;
+    static int8_t max_duty = 85;
+    static int8_t scaling_factor = 1; // to be defined
+
+    if (can_poll_receive_m162_temperature_set_3() == 0){
+        can_receive_m162_temperature_set_3();
+        current_temp = int(m162_temperature_set_3.d3_motor_temperature);
```

Comment:
any reason in particular on why these are ints?

Reply:
also you have to apply the scaling as described in the dbc to get the actual temperature value, not the decoded binary.